src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -299,6 +316,132 @@ public override void Initialize(AnalysisContext context)
                     }, SymbolKind.NamedType);
                 }
 
+                // Flag descriptor fields that are used to report compilation end diagnostics,
+                // but do not have the required 'WellKnownDiagnosticTags.CompilationEnd' custom tag.
+                // See https://github.com/dotnet/roslyn-analyzers/issues/6282 for details.
+                if (compilationEndContextType.GetMembers(DiagnosticWellKnownNames.ReportDiagnosticName).FirstOrDefault() is IMethodSymbol compilationEndReportDiagnosticMethod)
+                {
+                    var diagnosticCreateMethods = diagnosticType.GetMembers("Create").OfType<IMethodSymbol>()
+                        .Where(m => m.IsPublic() && m.Parameters.Length > 0 && SymbolEqualityComparer.Default.Equals(m.Parameters[0].Type, diagnosticDescriptorType))
+                        .ToImmutableHashSet(SymbolEqualityComparer.Default);
```

Comment:
This is 5 elements only. I'm not sure, but an array might have faster lookup than a hash set.

The difference might not be significant though.

Reply:
> The difference might not be significant though.

Agreed, I am going to leave it as-is for now.